drivers/SmartThings/matter-thermostat/src/init.lua

```diff
@@ -243,6 +370,54 @@ local function do_configure(driver, device)
     if #wind_eps > 0 then
       profile_name = profile_name .. "-wind"
     end
+
+    if #thermo_eps > 0 then
+      profile_name = profile_name .. "-thermostat"
+      if #humidity_eps > 0 and #fan_eps > 0 then
+        profile_name = profile_name .. "-humidity" .. "-fan"
+      elseif #humidity_eps > 0 then
+        profile_name = profile_name .. "-humidity"
+      elseif #fan_eps > 0 then
+        profile_name = profile_name .. "-fan"
+      end
+    end
+
+    if #heat_eps == 0 and #cool_eps == 0 then
+      device.log.warn_with({hub_logs=true}, "Thermostat does not support heating or cooling. No matching profile")
+      return
+    elseif #heat_eps > 0 and #cool_eps == 0 then
+      profile_name = profile_name .. "-heating-only"
+    elseif #cool_eps > 0 and #heat_eps == 0 then
+      profile_name = profile_name .. "-cooling-only"
+    end
+
+    -- TODO remove this in favor of reading Thermostat clusters AttributeList attribute
+    -- to determine support for ThermostatRunningState
+    -- Add nobattery profiles if updated
+    profile_name = profile_name .. "-nostate"
+
+    if #battery_eps == 0 then
+      profile_name = profile_name .. "-nobattery"
+    end
+
+    if #aqs_eps > 0 then
+      profile_name = profile_name .. "-aqs"
+    end
+
+    local meas_name, level_name = create_level_measurement_profile(device)
+
+    if meas_name ~= "" then
+      profile_name = profile_name .. meas_name .. "-meas"
+    end
+
+    if level_name ~= "" then
+      profile_name = profile_name .. level_name .. "-level"
+    end
+
+    if profile_name == "air-purifier-hepa-ac-wind-thermostat-humidity-fan-heating-only-nostate-nobattery-aqs-pm10-pm25-ch2o-meas-pm10-pm25-ch2o-no2-tvoc-level" then
```

Comment:
This should not longer be needed with the dossier profile name change

Reply:
yep, I'm going to remove it in the next update.

Reply:
removed